### PR TITLE
MODELIX-714 ModelQl query on model-server fails with "issue getting concept"

### PR DESCRIPTION
**Pull request by mhuster23**
These edge cases were not caught by the tests, because the client and server share the same `TypedLanguagesRegistry` as it is a a singleton.
In production the model-server would have no knowledge about the used languages, but it has them in our test.
If we wanted to test these cases automatically, we would probably have to test this with a real model-server, which would be quite expensive.

### CODE REVIEW

File: model-api/src/commonMain/kotlin/org/modelix/model/api/INode.kt [hunk 1]
```diff
@@ -324,7 +324,11 @@ fun INode.resolveProperty(role: String): IProperty {
 }
 
 fun INode.tryResolveChildLink(role: String): IChildLink? {
-    val c = this.concept ?: return null
+    val c = try {
+        this.concept ?: return null
+    } catch (e: RuntimeException) {
+        return null
```
Comment (slisson) [inline model-api/src/commonMain/kotlin/org/modelix/model/api/INode.kt:330]: This doesn't look like a proper fix. It just makes it harder to debug if there is a missing language. The relation between MODELIX-714 and this change is also not clear, so it's hard to propose an alternative solution.

Comment (languitar) [reply model-api/src/commonMain/kotlin/org/modelix/model/api/INode.kt:330]: To avoid such design discussion in the future, it would probably be good to specify what the `try` in the method names of the patched methods is intended to do. With this patch, `try` is extended to include the cases where a concept is unknown.

File: model-api-gen-gradle-test/ci.sh
Comment (languitar) [inline model-api-gen-gradle-test/ci.sh:1]: It's a bit unfortunate that a bug in the model-server is now effectively tested as part of the api-gen tests. Should we open up a follow-up ticket to create some real e2e tests?

File: model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt [hunk 2]
```diff
@@ -137,7 +137,7 @@ class ModelClientV2(
                 takeFrom(baseUrl)
                 appendPathSegments("repositories")
             }
-        }.bodyAsText().lines().map { RepositoryId(it) }
+        }.bodyAsText().lines().filter { it.isNotEmpty() }.map { RepositoryId(it) }
```
Comment (languitar) [inline model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt:140]: Why can empty lines happen?

**Review by languitar — APPROVED** [review]

File: model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt (hunk 2 above)
Comment (mhuster23) [reply model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt:140]: I explained it in the commit message.

File: model-api-gen-gradle-test/ci.sh
Comment (mhuster23) [reply model-api-gen-gradle-test/ci.sh:1]: [MODELIX-723](https://issues.modelix.org/issue/MODELIX-723/Create-end-to-end-tests-for-model-server)

File: model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt (hunk 2 above)
Comment (languitar) [reply model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt:140]: So the response includes a single newline and lines emits one empty line in that case?

Comment (mhuster23) [reply model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt:140]: I don't remember exactly anymore but I think the response body was empty. So `lines` returns the empty string.
